pkg/controllers/subnetset/subnetset_controller.go

```diff
@@ -75,7 +75,7 @@ func (r *SubnetSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 					return ResultRequeue, err
 				}
 				if obj.Spec.AccessMode == "" {
-					obj.Spec.AccessMode = v1alpha1.AccessMode(vpcNetworkConfig.DefaultSubnetAccessMode)
+					obj.Spec.AccessMode = v1alpha1.AccessMode(vpcNetworkConfig.DefaultPodSubnetAccessMode)
```

Comment:
We should only set DefaultPodSubnetAccessMode for default podSubnetSet.
We created default podSubnetSet  and default VMSubnetSetat  https://github.com/vmware-tanzu/nsx-operator/blob/main/pkg/controllers/namespace/namespace_controller.go#L119
Hence, we should set default VMSubnetset Accessmode as Private.

As for default podSubnetSet, we need to check it's obj'name is 
DefaultPodSubnetSet                string = "pod-default" AND obj.Spec.AccessMode  =""
to decide to set AccessMode from vpcNetworkConfig.DefaultPodSubnetAccessMode.

In summary:
1. Default VM subnetset Accessmode, set as private not be changed according to vpcNetworkConfig.DefaultPodSubnetAccessMode.
2. Default POD subnetset Accessmode, set Accessmode according to vpcNetworkConfig.DefaultPodSubnetAccessMode.
3. Other user created subnetset, set Accessmode as Private as default if user doesn't set this field.